_rules/device-motion-user-interface-7677a9.md

```diff
@@ -0,0 +1,355 @@
+---
+id: 7677a9
+name: Device motion can be operated by user interface
+rule_type: atomic
+description: |
+  This rule checks that functionality that can be operated by device motion can also be operated by user interface components.
+accessibility_requirements:
+  wcag21:2.5.4: # Motion Actuation (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+  - Accessibility tree
+acknowledgements:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [HTML document][] that has [functionality][] that can be operated by [device motion][].
+
+## Expectation
+
+For the test target, [device motion][] can also be operated by [user interface components][] and they should be [visible][] and [included in the accessibility tree][] with an [accessible name][] that is not empty ("").
+
+## Assumptions
+
+- This test assumes that the motion to operate [functionality][] is not used through an [accessibility supported][] interface
+- This test assumes that motion is not [essential][] for the [functionality][]
+
+## Accessibility Support
+
+- Device [orientation events][] may include sensitive data. Implementations must fire events only on [secure browsing contexts][]
+
+## Background
+
+- [Understanding Success Criterion 2.5.4: Motion Actuation](https://www.w3.org/WAI/WCAG21/Understanding/motion-actuation.html)
+- [G213: Provide conventional controls and an application setting for motion activated input](https://www.w3.org/WAI/WCAG21/Techniques/general/G213.html)
+- [DeviceOrientation Event Specification](https://www.w3.org/TR/orientation-event/)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The [HTML document][] has [functionality][] that can be operated by [device motion][] and can also be operated by [user interface components][] and they are [visible][] and [included in the accessibility tree][] with an [accessible name][] that is not empty ("").
+
+```html
+<html>
+  <head>
+    <title>Passed Example 1</title>
+    <script>
```

Comment:
Again, this script looks like it's identical in all Failed Examples except the first. It should be possible to refactor it in a test asset. Failed Example 1 can also be using it if this `activateSlider` has a `options` object with a `allowUIControl` or similar field.

Actually, both code for this rule and for "device motion can be disabled" can be put together. `activateSlider` will take an `options` object with two fields: `canBeDisabled` and `allowUIControl`. Given that `undefined` is a falsy value in JS, testing `options.canBeDisabled` will return `false` if not set. Hence, the calls need only to be:
- `activateSlider( {canBeDisabled: false} )` => other rule, Failed Example 1.
- `activateSlider( {canBeDisabled: true} )` => other rule, all other examples.
- `activateSlider( {allowUIControl: false} )` => this rule, Failed Example 1.
- `activateSlider( {allowUIControl: true} )` => this rule, all other examples.

(I'm in favour of keeping the explicit setting of the field to `false` even if not strictly necessary because this is making obvious what the example is illustrating.)

These calls are explicit enough that there is no need to perfectly remember the JS in order to understand the examples here. So we can safely hide the JS away.